src/extension/serializer.ts

```diff
@@ -733,26 +735,41 @@ export class GrpcSerializer extends SerializerBase {
     return await workspace.applyEdit(edit)
   }
 
+  protected notebookDataToProto(data: NotebookData): typeof this.protoNotebookType {
+    return Notebook.clone(data as any)
+  }
+
+  getSerializationRequest(obj: Object): SerializeRequest {
+    return <SerializeRequest>{ ...obj }
+  }
+
   protected async saveNotebook(
     data: NotebookData,
     // eslint-disable-next-line @typescript-eslint/no-unused-vars
     token: CancellationToken,
   ): Promise<Uint8Array> {
     const marshalFrontmatter = this.lifecycleIdentity === RunmeIdentity.ALL
 
-    const notebook = GrpcSerializer.marshalNotebook(data, { marshalFrontmatter })
+    const notebookProto = this.notebookDataToProto(data)
+    const notebook = Marshal.notebook(data, notebookProto, {
+      marshalFrontmatter,
+    }) as typeof this.protoNotebookType
 
     if (marshalFrontmatter) {
       data.metadata ??= {}
       data.metadata[RUNME_FRONTMATTER_PARSED] = notebook.frontmatter
     }
 
-    const cacheId = GrpcSerializer.getDocumentCacheId(data.metadata)
+    const cacheId = SerializerBase.getDocumentCacheId(data.metadata)
     this.notebookDataCache.set(cacheId as string, data)
 
-    const serialRequest = <SerializeRequest>{ notebook }
+    const serialRequest = this.getSerializationRequest({ notebook })
 
-    const cacheOutputs = this.cacheNotebookOutputs(notebook, cacheId)
+    const maskedNotebookProto = this.notebookDataToProto(data)
```

Comment:
Why are you masking the notebook here?

Reply:
a vestige of my attempt to abstract via composition.  see above.